tfs/src/main/java/hudson/plugins/tfs/model/AbstractHookEvent.java

```diff
@@ -113,7 +118,35 @@ static JSONObject fromResponseContributors(final List<GitStatus.ResponseContribu
                     return new TeamEventsEndpoint.ScheduledResponseContributor(project);
                 }
 
-                final TeamPushTrigger pushTrigger = TeamEventsEndpoint.findTrigger(job, TeamPushTrigger.class);
+                boolean shouldRun = false;
+                
+                if (job instanceof WorkflowJob)
+                {
+                    final FlowDefinition jobDef = ((WorkflowJob)job).getDefinition();
+                    if (jobDef instanceof CpsScmFlowDefinition)
+                    {
+                        final SCM jobSCM = ((CpsScmFlowDefinition) jobDef).getScm();
+                        if (jobSCM instanceof GitSCM)
+                        {
+                            final GitSCM gitJobSCM = (GitSCM)jobSCM;
+                            final URIish uri = gitCodePushedEventArgs.getRepoURIish();
+
+                            for (final UserRemoteConfig remoteConfig : gitJobSCM.getUserRemoteConfigs()) {
+                                final String jobRepoUrl = remoteConfig.getUrl();
+                                
+                                if (StringUtils.equals(jobRepoUrl, uri.toString())) {

```

Comment:
Didn't get a chance to fix this before it was merged. Included the fix in the PR of task: https://github.com/dotnet/core-eng/issues/830.